Assignment 5.Rmd

```diff
@@ -40,42 +41,57 @@ We want to see if we can build a decision tree to help teachers decide which stu
 
 #Visualize our outcome variable "score"
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
-D1$advice <- 
+D1$advice <- ifelse(D1$score>0.8,2,ifelse(D1$score<0.6,0,1))
+D1$advice
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- ctree(factor(advice) ~ 
+                       prior_prob_count + prior_percent_correct + hints, D1)
+summary(score_ctree)
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree)
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
+#0 - pay close attention to these students
+#1 - focus on these students sometimes
+#2 - great students
+
 
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv")
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
+D2$prediction <- predict(score_ctree,D2)
 
 ``` 
+
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
 
+```{r}
+mean(ifelse(D2$prediction == "no action", 1, 0))
+# So the accuracy of the model on the new data would be 58%.
+```
+All students in the new sample achieved a score of 1, 58% of students from this new data set would have recieved no action from the teacher. 
```

Comment:
What does this say about the model? Does it overfit or is it more generalizable? Nonetheless, great job overall!